db-scripts/src/main/resources/cgds.sql

```diff
@@ -676,16 +692,16 @@ CREATE TABLE `sample_cna_event` (
 -- --------------------------------------------------------
 CREATE TABLE `copy_number_seg` (
   `SEG_ID` int(255) NOT NULL auto_increment,
-  `CANCER_STUDY_ID` int(11) NOT NULL,
+  `GENETIC_PROFILE_ID` int(11) NOT NULL,
```

Comment:
this is a significant change for this table. Validation and data loading scripts will have to be changed. Backend services and probably frontend code as well.

Reply:
I would propose a PR that isolates just this change. The the current PR can more easily be built on top of that.

Reply:
@pieterlukasse @jjgao 
support multiple genomes at the profiling level (replacing cancer_study_id by genetic_profile_id) is one major change introduced by this PR. It affects backend services ,frontend react components, and importing scripts. 
But, JJ made a good point, having all multiple genome related changes in one PR will save us from multiple database migrations, which are also quite painful. 
I would say by planning carefully, we could push through all changes at once. Let us have a meeting discuss more about this sometime next week. I will draft a plan on things need to be changed.

Reply:
I am still not sure which way would be the best. I am also worried that this pr got too big and difficult to merge.

Reply:
how about we go through the list of things need to be changed, and decide from there? 
Two major changes introduced in this PR:
1. adding a new reference_genome_gene table, the changes are mainly made to the the persistence-mybatis component (SQL statements with ab extra INNER JOIN, the return set kept as same as before for not interfering with existing codes) 
2. replacing cancer_study_Id with genetic_profile_id, that affects the web APIs, front-end asyn APIs (minor, just need to include an additional genetic_profile_id when calling API to retrieve copy number segments). 

Big work comes from correcting those **bird nest** import and test scripts to work with new table structures,  and fixing those hard coded test cases. Excluding the first part really not help much I would say.

Reply:
@khzhu so, what I am proposing is splitting up 1 and 2 into separate PRs. So starting with 2 and then doing 1. Do you think you can split this up without too much effort? It would help in the review and testing effort.